src/consensus/encode.rs

```diff
@@ -597,6 +644,33 @@ pub(crate) fn consensus_encode_with_size<S: io::Write>(data: &[u8], mut s: S) ->
 }
 
 
+struct ReadBytesFromFiniteReaderOpts {
+    len: usize,
+    chunk_size: usize,
+}
+
+/// Read `len` bytes from reader, where `len` could potentially be malicious
+///
+/// This function relies on `reader` being bound in amount of data
+/// it returns for OOM protection. See [`Decodable::consensus_decode_from_finite_reader`].
+#[inline]
+fn read_bytes_from_finite_reader<D: io::Read>(mut d: D, mut opts: ReadBytesFromFiniteReaderOpts) -> Result<Vec<u8>, Error> {
+    let mut ret = vec![];
+
+    debug_assert_ne!(opts.chunk_size, 0);
```

Comment:
A question please since we have been discussing debug asserts a bit but I'm still not totally understanding when to use what. This debug statement is defensive programming against insane input with chunck size 0 causing an infinite loop, right? What made you choose to use debug assert instead of either 
A. a regular asset
B.  returning an error

Reply:
I wasn't really sure myself what to do. `chunk_size` should always be a constant, and this whole function is kind of internal, so I thought checking in `debug_` is good enough to spot some silly mistake. I'm still not sure about it. I wish I could make `chunk_size` a `const` of some kind.

Reply:
I just turned it into `assert_ne`. Compiler can probably optimize it anyway.